pkg/controller/kogitoinfra/kafka/kafka_reconciler.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"fmt"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	kafkav1beta1 "github.com/kiegroup/kogito-cloud-operator/pkg/apis/kafka/v1beta1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// KafkaResource implementation of KogitoInfraResource
+type KafkaResource struct {
+}
+
+// GetWatchedObjects provide list of object that needs to be watched to maintain Kafka kogitoInfra resource
+func GetWatchedObjects() []framework.WatchedObjects {
+	return []framework.WatchedObjects{
+		{
+			GroupVersion: kafkav1beta1.SchemeGroupVersion,
+			AddToScheme:  kafkav1beta1.SchemeBuilder.AddToScheme,
+			Objects:      []runtime.Object{&kafkav1beta1.Kafka{}},
+		},
+	}
+}
+
+// Reconcile reconcile Kogito infra object
+func (k *KafkaResource) Reconcile(client *client.Client, instance *v1alpha1.KogitoInfra, scheme *runtime.Scheme) (requeue bool, resultErr error) {
+
+	var kafkaInstance *kafkav1beta1.Kafka
+
+	// Step 1: check whether user has provided custom infinispan instance reference
+	isCustomReferenceProvided := len(instance.Spec.Resource.Name) > 0
+	if isCustomReferenceProvided {
+		log.Debugf("Custom kafka instance reference is provided")
+		resourceName := instance.Spec.Resource.Name
+		resourceNameSpace := instance.Spec.Resource.Namespace
```

Comment:
Same here about the temps

Reply:
Similarly we are setting below properties for kafka :
```
        kafkaProperties := &instance.Status.Kafka.KafkaProperties
	kafkaProperties.ExternalURI = uri
```

Reply:
I had removed these temp